storage-node/src/services/webApi/controllers/filesApi.ts

```diff
@@ -34,14 +38,19 @@ export async function getFile(
   res: express.Response<unknown, AppConfig>,
   next: express.NextFunction
 ): Promise<void> {
-  try {
-    await pinDataObjectIdToCache(req.params.id)
+  const { id } = req.params
+  const dataObjectId = new BN(id).toString()
 
-    const dataObjectId = new BN(req.params.id)
-    const uploadsDir = res.locals.uploadsDir
+  if (!(await getDataObjectIdFromCache(dataObjectId))) {
+    sendResponseWithError(res, next, new WebApiError('File Not Found', 404), 'files')
+    return
+  }
```

Comment:
Wouldn't this be redundant, since now `pinDataObjectIdToCache` ensures that ID exists before pinning it? If this is added to return a better error message in the response, then we can provide this error in as an argument of `asset` in `pinDataObjectIdToCache` function itself. e.g.

```ts
  assert(idCache.has(dataObjectId), new WebApiError('File Not Found', 404))

```

Reply:
We can also pass the `Error` as a second optional argument to the `pinDataObjectIdToCache` function

Reply:
Remember assertions don't throw and it is intended to fail the process completely. Meaning the caller needs to make sure they are calling at the right time. So if these new assertions fail it means our code is not correct.

Are you suggesting we should be returning/throwing an error/exception instead of these assertions?

Reply:
I picked assertions because I was treating the user of the idCache as "our own" code. So its not like a library.
With reference to https://softwareengineering.stackexchange.com/questions/15515/when-to-use-assertions-and-when-to-use-exceptions#15518 

Let me know what you think.

Reply:
> Remember assertions don't throw and it is intended to fail the process completely.

Maybe I am misunderstanding, but In javascript, the `assert` module throws the Error or type [AssertionError](https://nodejs.org/api/assert.html#class-assertassertionerror), and we can actually pass a custom error too, that should be thrown in case assertion fails.

Reply:
You are absolutely correct. And given that is true I think I might be introducing more problems.
Let me think this through and come back to it.